modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java

```diff
@@ -126,9 +126,9 @@
     /** Last exchange future. */
     private volatile GridDhtPartitionsExchangeFuture lastExchangeFut;
 
-    /** Futures involved in the last rebalance. For statistics. */
+    /** Total statistics of rebalance. */
     @GridToStringExclude
-    private final Collection<RebalanceFuture> lastStatFutures = new ConcurrentLinkedQueue<>();
+    @Nullable private final CacheGroupTotalRebalanceStatistics totalRebStat;
```

Comment:
What's the point for keeping total statistics for every group separately if they are going to merged anyway?

Reply:
I didn't find a better place, can you suggest it?

Reply:
In personal correspondence, we decided don't change it.